native/src/image.rs

```diff
@@ -85,55 +85,38 @@ impl Hash for Handle {
     }
 }
 
-/// Wrapper around raw image data.
+/// A wrapper around raw image data.
 ///
-/// Behaves like a `&[u8]`.
-pub struct ImageBytes(Box<dyn ImageBytesTrait>);
+/// It behaves like a `&[u8]`.
+#[derive(Clone)]
+pub struct Bytes(Arc<dyn AsRef<[u8]> + Send + Sync + 'static>);
 
-trait ImageBytesTrait: AsRef<[u8]> + Send + Sync + 'static {
-    fn clone_boxed(&self) -> Box<dyn ImageBytesTrait>;
-}
-
-impl<T: AsRef<[u8]> + Clone + Send + Sync + 'static> ImageBytesTrait for T {
-    fn clone_boxed(&self) -> Box<dyn ImageBytesTrait> {
-        Box::new(self.clone())
-    }
-}
-
-impl ImageBytes {
-    /// Creates a new `ImageBytes` around `data`.
-    pub fn new(
-        data: impl AsRef<[u8]> + Clone + Send + Sync + 'static,
-    ) -> ImageBytes {
-        Self(Box::new(data))
-    }
-}
-
-impl Clone for ImageBytes {
-    fn clone(&self) -> Self {
-        ImageBytes(self.0.clone_boxed())
+impl Bytes {
+    /// Creates new [`Bytes`] around `data`.
+    pub fn new(data: impl AsRef<[u8]> + Clone + Send + Sync + 'static) -> Self {
```

Comment:
You don't really need the `Clone` here anymore as the `Arc` already gives you that :)
Do you want me to submit a PR for removing the trait bound or will you take care of the change?

Reply:
Good catch. Just opened #1717.

Reply:
Thanks!